### DISCUSSION TITLE
Spatial Join and Duplicates?

### DISCUSSION THREAD

**Discussion by edchanonline** [earthlab-education/earth-analytics-02-python-2024#6] [Q&A01 Urban Greenspace | Cloud Data] [answered]
Hey everyone! I am trying to understand if I am missing a step during spatial join. 

Before joining, I made sure the projections between my dataframes (`tracts` and `neigh_gdf` ) are both matching (projected to `EPSG:4269`).

I filtered my chicago boundary to `['Logan Square', 'Lincoln Park']` for testing. 

I am joining with the `intersects` predicate, but I am producing a dataframe with some duplicate shapes. 

```
chi_tracts = gpd.sjoin(tracts, neigh_gdf, predicate='intersects')
```
![Screenshot 2024-01-24 at 11 38 02 PM](https://github.com/earthlab-education/earth-analytics-02-python-2024/assets/4795392/eb384262-a595-4db4-b7d3-7a675d7e9369)

I think `intersects` should be on the right track as `within` and `contains` would be overly excluding some area of interests. 

Are you doing anything to remove duplicates? Did you simply remove duplicates by checking for repeating `TRACTCE` IDs? I want to make sure my NAIP data requests are unique so I don't run into processing 1500+ stac item searches vs ~800 stac item searches.

**Comment by eculler** [comment, marked as answer]
@eggvoice hm, it would appear that this census track intersects both neighborhoods. This shouldn't be an issue for you if you are using only the single Chicago boundary. I think it would be a lot of extra work to, for example, select the neighborhood that contains the largest portion of the Census tract.

For testing, you could consider working with a single neighborhood, or even set up your code to only run the first few census tracts, skipping the neighborhood altogether.

**Reply by edchanonline** [reply]
The answer is to either 

1) Dissolve the boundaries between neighborhoods

```
chi_gdf = chi_gdf.dissolve()
```

2) Use the single Chicago boundary dataset (not the neighborhoods one)
